diff -Nur snfit-2.2.2b.orig/configure.ac snfit-2.2.2b/configure.ac
--- snfit-2.2.2b.orig/configure.ac	2011-02-08 22:01:37.000000000 +0000
+++ snfit-2.2.2b/configure.ac	2011-02-08 22:01:46.000000000 +0000
@@ -4,75 +4,61 @@
 AC_PREREQ(2.59)
 AC_INIT(snfit, 2.2.2b, julien.guy@lpnhe.in2p3.fr)
 AC_CONFIG_SRCDIR([src])
-AC_CONFIG_HEADER([config.h])
+AC_CONFIG_HEADER([src/snfit_config.h])
+AC_CONFIG_MACRO_DIR([m4])
+AC_CONFIG_AUX_DIR([autoconf])
 AM_INIT_AUTOMAKE
 
 # Checks for programs.
-AC_PROG_LIBTOOL
+AC_DISABLE_STATIC
 AC_PROG_CXX
 AC_PROG_CC
 AC_PROG_F77
+AC_F77_LIBRARY_LDFLAGS
+AC_PROG_LIBTOOL
 
-# Checks for libraries.
-AC_CHECK_LIB([g2c], [main],, AC_CHECK_LIB([f2c],[main],,))
-AC_CHECK_LIB([blas], [main])
-AC_CHECK_LIB([lapack], [main])
-
-# enable gfortran for mac intel
-AC_ARG_ENABLE(gfortran,
-[AC_HELP_STRING([--enable-gfortran],[Say yes for mac intel ])],
-[ gfortran=true
-AC_CHECK_LIB([gfortran],[main],,
-	[AC_MSG_ERROR("Cannot find library gfortran.")])
-  GFOR='-DgFortran'
-  AC_SUBST(GFOR)
-],
-[ gfortran=false 
-  GFOR=''
-  AC_SUBST(GFOR)
-])
-
-#cfitsio
-AC_ARG_WITH([cfitsio-libdir], 
-[AS_HELP_STRING([--with-cfitsio-libdir],[use ciftsio])],	       
-[cfitsiolibarg="$withval"],
-[])
-AC_ARG_WITH([cfitsio-incdir], 
-[AS_HELP_STRING([--with-cfitsio-incdir],[use ciftsio])],	       
-[cfitsioincarg="$withval"],
-[])
-
-if test $cfitsiolibarg; then
-     if test $cfitsioincarg; then
-     
-	## check lib
-	AC_CHECK_LIB([cfitsio],[main],,
-		AC_MSG_ERROR("cannot link with -L$cfitsiolibarg -lcfitsio"),
-		[-L$cfitsiolibarg] )
-	
-	AC_CHECK_HEADERS([$cfitsioincarg/fitsio.h])
-
-	AC_SUBST(CFITSIOCPPFLAGS,["-DUSECFITSIO -I$cfitsioincarg"])
-	AC_SUBST(CFITSIOLDFLAGS,["-R$cfitsiolibarg -L$cfitsiolibarg -lcfitsio"])
-	
-
-     else
-	AC_MSG_ERROR("need both --with-cfitsio-libdir and --with-cfitsio-incdir or none")
-
-     fi
 
+if test x${F77:=${FC}} = xgfortran; then
+      cfortran_def=gFortran
+elif test x$F77 = xifort ; then
+      cfortran_def=INTEL_COMPILER
+elif test x$F77 = xg77 ; then
+      cfortran_def=g77Fortran
+elif test x$F77 = xpgf77 ; then
+      cfortran_def=pgiFortran
+else
+      cfortran_def=f2cFortran
 fi
+CFORTRAN_CFLAGS="-D$cfortran_def"
+AC_SUBST(CFORTRAN_CFLAGS)
 
-CXXFLAGS="-g -Wall -O3"
-CPPFLAGS="-g -Wall -O3"
-CFLAGS="-g -Wall -O3"
-FFLAGS="-g -O3"
+# Checks for libraries.
+AX_LAPACK(, [AC_MSG_ERROR([blas or lapack was not found.
+*** Please get blas and lapack installed, or use the options
+*** --with-blas or --with-lapack.])
+])
 
+# Check for optional cfitsio
+AC_MSG_CHECKING([whether to build cfitsio support])
+AC_ARG_WITH([cfitsio],
+   AS_HELP_STRING([--without-cfitsio], [do not build cfitsio support]),
+   [with_cfitsio=$withval],
+   [with_cfitsio=yes])
+if test x"$with_cfitsio" = "xyes"; then
+    AC_MSG_RESULT([yes])
+    AX_CFITSIO(, [AC_MSG_ERROR([cfitsio library was not found.
+   *** Please install cfitsio or use the options:
+   *** --with-cfitsio-includedir and --with-cfitsio-libdir.])
+   ])
+else
+    AC_MSG_RESULT([no])
+fi
 
 # Checks for header files.
 AC_HEADER_DIRENT
 AC_HEADER_STDC
 
 AC_CONFIG_FILES([Makefile
+		 snfit.pc
                  src/Makefile])
 AC_OUTPUT
diff -Nur snfit-2.2.2b.orig/m4/ax_blas.m4 snfit-2.2.2b/m4/ax_blas.m4
--- snfit-2.2.2b.orig/m4/ax_blas.m4	1970-01-01 01:00:00.000000000 +0100
+++ snfit-2.2.2b/m4/ax_blas.m4	2011-02-08 22:01:46.000000000 +0000
@@ -0,0 +1,197 @@
+# ===========================================================================
+#          http://www.gnu.org/software/autoconf-archive/ax_blas.html
+# ===========================================================================
+#
+# SYNOPSIS
+#
+#   AX_BLAS([ACTION-IF-FOUND[, ACTION-IF-NOT-FOUND]])
+#
+# DESCRIPTION
+#
+#   This macro looks for a library that implements the BLAS linear-algebra
+#   interface (see http://www.netlib.org/blas/). On success, it sets the
+#   BLAS_LIBS output variable to hold the requisite library linkages.
+#
+#   To link with BLAS, you should link with:
+#
+#     $BLAS_LIBS $LIBS $FLIBS
+#
+#   in that order. FLIBS is the output variable of the
+#   AC_F77_LIBRARY_LDFLAGS macro (called if necessary by AX_BLAS), and is
+#   sometimes necessary in order to link with F77 libraries. Users will also
+#   need to use AC_F77_DUMMY_MAIN (see the autoconf manual), for the same
+#   reason.
+#
+#   Many libraries are searched for, from ATLAS to CXML to ESSL. The user
+#   may also use --with-blas=<lib> in order to use some specific BLAS
+#   library <lib>. In order to link successfully, however, be aware that you
+#   will probably need to use the same Fortran compiler (which can be set
+#   via the F77 env. var.) as was used to compile the BLAS library.
+#
+#   ACTION-IF-FOUND is a list of shell commands to run if a BLAS library is
+#   found, and ACTION-IF-NOT-FOUND is a list of commands to run it if it is
+#   not found. If ACTION-IF-FOUND is not specified, the default action will
+#   define HAVE_BLAS.
+#
+# LICENSE
+#
+#   Copyright (c) 2008 Steven G. Johnson <stevenj@alum.mit.edu>
+#
+#   This program is free software: you can redistribute it and/or modify it
+#   under the terms of the GNU General Public License as published by the
+#   Free Software Foundation, either version 3 of the License, or (at your
+#   option) any later version.
+#
+#   This program is distributed in the hope that it will be useful, but
+#   WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+#   Public License for more details.
+#
+#   You should have received a copy of the GNU General Public License along
+#   with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+#   As a special exception, the respective Autoconf Macro's copyright owner
+#   gives unlimited permission to copy, distribute and modify the configure
+#   scripts that are the output of Autoconf when processing the Macro. You
+#   need not follow the terms of the GNU General Public License when using
+#   or distributing such scripts, even though portions of the text of the
+#   Macro appear in them. The GNU General Public License (GPL) does govern
+#   all other use of the material that constitutes the Autoconf Macro.
+#
+#   This special exception to the GPL applies to versions of the Autoconf
+#   Macro released by the Autoconf Archive. When you make and distribute a
+#   modified version of the Autoconf Macro, you may extend this special
+#   exception to the GPL to apply to your modified version as well.
+
+#serial 8
+
+AU_ALIAS([ACX_BLAS], [AX_BLAS])
+AC_DEFUN([AX_BLAS], [
+AC_PREREQ(2.50)
+AC_REQUIRE([AC_F77_LIBRARY_LDFLAGS])
+ax_blas_ok=no
+
+AC_ARG_WITH(blas,
+	[AS_HELP_STRING([--with-blas=<lib>], [use BLAS library <lib>])])
+case $with_blas in
+	yes | "") ;;
+	no) ax_blas_ok=disable ;;
+	-* | */* | *.a | *.so | *.so.* | *.o) BLAS_LIBS="$with_blas" ;;
+	*) BLAS_LIBS="-l$with_blas" ;;
+esac
+
+# Get fortran linker names of BLAS functions to check for.
+AC_F77_FUNC(sgemm)
+AC_F77_FUNC(dgemm)
+
+ax_blas_save_LIBS="$LIBS"
+LIBS="$LIBS $FLIBS"
+
+# First, check BLAS_LIBS environment variable
+if test $ax_blas_ok = no; then
+if test "x$BLAS_LIBS" != x; then
+	save_LIBS="$LIBS"; LIBS="$BLAS_LIBS $LIBS"
+	AC_MSG_CHECKING([for $sgemm in $BLAS_LIBS])
+	AC_TRY_LINK_FUNC($sgemm, [ax_blas_ok=yes], [BLAS_LIBS=""])
+	AC_MSG_RESULT($ax_blas_ok)
+	LIBS="$save_LIBS"
+fi
+fi
+
+# BLAS linked to by default?  (happens on some supercomputers)
+if test $ax_blas_ok = no; then
+	save_LIBS="$LIBS"; LIBS="$LIBS"
+	AC_CHECK_FUNC($sgemm, [ax_blas_ok=yes])
+	LIBS="$save_LIBS"
+fi
+
+# BLAS in ATLAS library? (http://math-atlas.sourceforge.net/)
+if test $ax_blas_ok = no; then
+	AC_CHECK_LIB(atlas, ATL_xerbla,
+		[AC_CHECK_LIB(f77blas, $sgemm,
+		[AC_CHECK_LIB(cblas, cblas_dgemm,
+			[ax_blas_ok=yes
+			 BLAS_LIBS="-lcblas -lf77blas -latlas"],
+			[], [-lf77blas -latlas])],
+			[], [-latlas])])
+fi
+
+# BLAS in PhiPACK libraries? (requires generic BLAS lib, too)
+if test $ax_blas_ok = no; then
+	AC_CHECK_LIB(blas, $sgemm,
+		[AC_CHECK_LIB(dgemm, $dgemm,
+		[AC_CHECK_LIB(sgemm, $sgemm,
+			[ax_blas_ok=yes; BLAS_LIBS="-lsgemm -ldgemm -lblas"],
+			[], [-lblas])],
+			[], [-lblas])])
+fi
+
+# BLAS in Intel MKL library?
+if test $ax_blas_ok = no; then
+	AC_CHECK_LIB(mkl, $sgemm, [ax_blas_ok=yes;BLAS_LIBS="-lmkl"])
+fi
+
+# BLAS in Apple vecLib library?
+if test $ax_blas_ok = no; then
+	save_LIBS="$LIBS"; LIBS="-framework vecLib $LIBS"
+	AC_CHECK_FUNC($sgemm, [ax_blas_ok=yes;BLAS_LIBS="-framework vecLib"])
+	LIBS="$save_LIBS"
+fi
+
+# BLAS in Alpha CXML library?
+if test $ax_blas_ok = no; then
+	AC_CHECK_LIB(cxml, $sgemm, [ax_blas_ok=yes;BLAS_LIBS="-lcxml"])
+fi
+
+# BLAS in Alpha DXML library? (now called CXML, see above)
+if test $ax_blas_ok = no; then
+	AC_CHECK_LIB(dxml, $sgemm, [ax_blas_ok=yes;BLAS_LIBS="-ldxml"])
+fi
+
+# BLAS in Sun Performance library?
+if test $ax_blas_ok = no; then
+	if test "x$GCC" != xyes; then # only works with Sun CC
+		AC_CHECK_LIB(sunmath, acosp,
+			[AC_CHECK_LIB(sunperf, $sgemm,
+        			[BLAS_LIBS="-xlic_lib=sunperf -lsunmath"
+                                 ax_blas_ok=yes],[],[-lsunmath])])
+	fi
+fi
+
+# BLAS in SCSL library?  (SGI/Cray Scientific Library)
+if test $ax_blas_ok = no; then
+	AC_CHECK_LIB(scs, $sgemm, [ax_blas_ok=yes; BLAS_LIBS="-lscs"])
+fi
+
+# BLAS in SGIMATH library?
+if test $ax_blas_ok = no; then
+	AC_CHECK_LIB(complib.sgimath, $sgemm,
+		     [ax_blas_ok=yes; BLAS_LIBS="-lcomplib.sgimath"])
+fi
+
+# BLAS in IBM ESSL library? (requires generic BLAS lib, too)
+if test $ax_blas_ok = no; then
+	AC_CHECK_LIB(blas, $sgemm,
+		[AC_CHECK_LIB(essl, $sgemm,
+			[ax_blas_ok=yes; BLAS_LIBS="-lessl -lblas"],
+			[], [-lblas $FLIBS])])
+fi
+
+# Generic BLAS library?
+if test $ax_blas_ok = no; then
+	AC_CHECK_LIB(blas, $sgemm, [ax_blas_ok=yes; BLAS_LIBS="-lblas"])
+fi
+
+AC_SUBST(BLAS_LIBS)
+
+LIBS="$ax_blas_save_LIBS"
+
+# Finally, execute ACTION-IF-FOUND/ACTION-IF-NOT-FOUND:
+if test x"$ax_blas_ok" = xyes; then
+        ifelse([$1],,AC_DEFINE(HAVE_BLAS,1,[Define if you have a BLAS library.]),[$1])
+        :
+else
+        ax_blas_ok=no
+        $2
+fi
+])dnl AX_BLAS
diff -Nur snfit-2.2.2b.orig/m4/ax_cfitsio.m4 snfit-2.2.2b/m4/ax_cfitsio.m4
--- snfit-2.2.2b.orig/m4/ax_cfitsio.m4	1970-01-01 01:00:00.000000000 +0100
+++ snfit-2.2.2b/m4/ax_cfitsio.m4	2011-02-08 22:01:46.000000000 +0000
@@ -0,0 +1,105 @@
+#
+# SYNOPSIS
+#
+#   AX_CFITSIO([ACTION-IF-FOUND[, ACTION-IF-NOT-FOUND]])
+#
+# DESCRIPTION
+#
+#   This macro looks for the cfitsio library (see
+#   http://heasarc.gsfc.nasa.gov/fitsio/). On success, it sets the
+#   CFITSIO_LIBS and CFITSIO_CFLAGS output variables to hold the
+#   requisite library linkages. It first look for a pkg-config file
+#   before trying to compile and link with cfitsio.
+#
+#   To link with CFITSIO, you should link with:
+#
+#     $CFITSIO_LIBS $LIBS
+#
+#   ACTION-IF-FOUND is a list of shell commands to run if the CFITSIO
+#   library is found, and ACTION-IF-NOT-FOUND is a list of commands to
+#   run it if it is not found. If ACTION-IF-FOUND is not specified, the
+#   default action will define the processor macro HAVE_CFITSIO. 
+#
+# LICENSE
+#
+#   Copyright (c) 2010 Sebastien Fabbro <sfabbro@uvic.ca>
+#
+#   This program is free software: you can redistribute it and/or modify it
+#   under the terms of the GNU General Public License as published by the
+#   Free Software Foundation, either version 3 of the License, or (at your
+#   option) any later version.
+#
+#   This program is distributed in the hope that it will be useful, but
+#   WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+#   Public License for more details.
+#
+#   You should have received a copy of the GNU General Public License along
+#   with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+#   As a special exception, the respective Autoconf Macro's copyright owner
+#   gives unlimited permission to copy, distribute and modify the configure
+#   scripts that are the output of Autoconf when processing the Macro. You
+#   need not follow the terms of the GNU General Public License when using
+#   or distributing such scripts, even though portions of the text of the
+#   Macro appear in them. The GNU General Public License (GPL) does govern
+#   all other use of the material that constitutes the Autoconf Macro.
+#
+#   This special exception to the GPL applies to versions of the Autoconf
+#   Macro released by the Autoconf Archive. When you make and distribute a
+#   modified version of the Autoconf Macro, you may extend this special
+#   exception to the GPL to apply to your modified version as well.
+
+
+AC_DEFUN([AX_CFITSIO],[
+
+ax_cfitsio_ok=no
+
+PKG_CHECK_MODULES(CFITSIO, cfitsio, [ax_cfitsio_ok=yes], [ax_cfitsio_ok=no])
+
+if test x"$ax_cfitsio_ok" = x"no"; then
+   # if neither pkg-config file found, nor CFITSIO_LIBS defined
+   if test x$CFITSIO_LIBS = x ; then
+      CFITSIO_LIBS="-lcfitsio -lm"
+      AC_ARG_WITH([cfitsio-libdir],
+	 [  --with-cfitsio-libdir=DIR directory where the library was installed],
+	 [CFITSIO_LIBS="-L$withval $CFITSIO_LIBS"], )
+   fi
+   ax_cfitsio_lib_ok=no
+   LIBS_sav="$LIBS"
+   LIBS="$LIBS $CFITSIO_LIBS"
+   AC_CHECK_LIB([cfitsio], [ffopen], [ax_cfitsio_lib_ok=yes], [AC_MSG_WARN([  *** cfitsio library not found])])
+
+   # if neither pkg-config file found, nor CFITSIO_CFLAGS defined
+   if test x$CFITSIO_CFLAGS = x ; then
+      CFITSIO_CFLAGS=""
+      AC_ARG_WITH(cfitsio-includedir,
+	 [  --with-cfitsio-includedir=DIR directory where the headers were installed],
+	 [CFITSIO_CFLAGS="-I$withval"], )
+   fi
+   ax_cfitsio_hdr_ok=no
+   CPPFLAGS_sav="$CPPFLAGS"
+   CPPFLAGS="$CPPFLAGS $CFITSIO_CFLAGS"
+   AC_CHECK_HEADER([fitsio.h],
+      [ax_cfitsio_hdr_ok=yes],
+      [AC_MSG_WARN([  *** cfitsio headers not found.])])
+
+   if test x$ax_cfitsio_lib_ok = xyes -a x$ax_cfitsio_hdr_ok = xyes; then
+      ax_cfitsio_ok=yes
+   fi
+   AC_SUBST(CFITSIO_CFLAGS)
+   AC_SUBST(CFITSIO_LIBS)
+   LIBS="$LIBS_sav"
+   CPPFLAGS="$CPPFLAGS_sav"
+fi
+
+
+# Finally, execute ACTION-IF-FOUND/ACTION-IF-NOT-FOUND:
+if test x"$ax_cfitsio_ok" = x"yes"; then
+        ifelse([$1],,AC_DEFINE(HAVE_CFITSIO, [1], [Define if you have CFITSIO library.]),[$1])
+        :
+else
+        ax_cfitsio_ok=no
+        $2
+fi
+])
diff -Nur snfit-2.2.2b.orig/m4/ax_lapack.m4 snfit-2.2.2b/m4/ax_lapack.m4
--- snfit-2.2.2b.orig/m4/ax_lapack.m4	1970-01-01 01:00:00.000000000 +0100
+++ snfit-2.2.2b/m4/ax_lapack.m4	2011-02-08 22:01:46.000000000 +0000
@@ -0,0 +1,131 @@
+# ===========================================================================
+#         http://www.gnu.org/software/autoconf-archive/ax_lapack.html
+# ===========================================================================
+#
+# SYNOPSIS
+#
+#   AX_LAPACK([ACTION-IF-FOUND[, ACTION-IF-NOT-FOUND]])
+#
+# DESCRIPTION
+#
+#   This macro looks for a library that implements the LAPACK linear-algebra
+#   interface (see http://www.netlib.org/lapack/). On success, it sets the
+#   LAPACK_LIBS output variable to hold the requisite library linkages.
+#
+#   To link with LAPACK, you should link with:
+#
+#     $LAPACK_LIBS $BLAS_LIBS $LIBS $FLIBS
+#
+#   in that order. BLAS_LIBS is the output variable of the AX_BLAS macro,
+#   called automatically. FLIBS is the output variable of the
+#   AC_F77_LIBRARY_LDFLAGS macro (called if necessary by AX_BLAS), and is
+#   sometimes necessary in order to link with F77 libraries. Users will also
+#   need to use AC_F77_DUMMY_MAIN (see the autoconf manual), for the same
+#   reason.
+#
+#   The user may also use --with-lapack=<lib> in order to use some specific
+#   LAPACK library <lib>. In order to link successfully, however, be aware
+#   that you will probably need to use the same Fortran compiler (which can
+#   be set via the F77 env. var.) as was used to compile the LAPACK and BLAS
+#   libraries.
+#
+#   ACTION-IF-FOUND is a list of shell commands to run if a LAPACK library
+#   is found, and ACTION-IF-NOT-FOUND is a list of commands to run it if it
+#   is not found. If ACTION-IF-FOUND is not specified, the default action
+#   will define HAVE_LAPACK.
+#
+# LICENSE
+#
+#   Copyright (c) 2009 Steven G. Johnson <stevenj@alum.mit.edu>
+#
+#   This program is free software: you can redistribute it and/or modify it
+#   under the terms of the GNU General Public License as published by the
+#   Free Software Foundation, either version 3 of the License, or (at your
+#   option) any later version.
+#
+#   This program is distributed in the hope that it will be useful, but
+#   WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+#   Public License for more details.
+#
+#   You should have received a copy of the GNU General Public License along
+#   with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+#   As a special exception, the respective Autoconf Macro's copyright owner
+#   gives unlimited permission to copy, distribute and modify the configure
+#   scripts that are the output of Autoconf when processing the Macro. You
+#   need not follow the terms of the GNU General Public License when using
+#   or distributing such scripts, even though portions of the text of the
+#   Macro appear in them. The GNU General Public License (GPL) does govern
+#   all other use of the material that constitutes the Autoconf Macro.
+#
+#   This special exception to the GPL applies to versions of the Autoconf
+#   Macro released by the Autoconf Archive. When you make and distribute a
+#   modified version of the Autoconf Macro, you may extend this special
+#   exception to the GPL to apply to your modified version as well.
+
+#serial 7
+
+AU_ALIAS([ACX_LAPACK], [AX_LAPACK])
+AC_DEFUN([AX_LAPACK], [
+AC_REQUIRE([AX_BLAS])
+ax_lapack_ok=no
+
+AC_ARG_WITH(lapack,
+        [AS_HELP_STRING([--with-lapack=<lib>], [use LAPACK library <lib>])])
+case $with_lapack in
+        yes | "") ;;
+        no) ax_lapack_ok=disable ;;
+        -* | */* | *.a | *.so | *.so.* | *.o) LAPACK_LIBS="$with_lapack" ;;
+        *) LAPACK_LIBS="-l$with_lapack" ;;
+esac
+
+# Get fortran linker name of LAPACK function to check for.
+AC_F77_FUNC(cheev)
+
+# We cannot use LAPACK if BLAS is not found
+if test "x$ax_blas_ok" != xyes; then
+        ax_lapack_ok=noblas
+        LAPACK_LIBS=""
+fi
+
+# First, check LAPACK_LIBS environment variable
+if test "x$LAPACK_LIBS" != x; then
+        save_LIBS="$LIBS"; LIBS="$LAPACK_LIBS $BLAS_LIBS $LIBS $FLIBS"
+        AC_MSG_CHECKING([for $cheev in $LAPACK_LIBS])
+        AC_TRY_LINK_FUNC($cheev, [ax_lapack_ok=yes], [LAPACK_LIBS=""])
+        AC_MSG_RESULT($ax_lapack_ok)
+        LIBS="$save_LIBS"
+        if test $ax_lapack_ok = no; then
+                LAPACK_LIBS=""
+        fi
+fi
+
+# LAPACK linked to by default?  (is sometimes included in BLAS lib)
+if test $ax_lapack_ok = no; then
+        save_LIBS="$LIBS"; LIBS="$LIBS $BLAS_LIBS $FLIBS"
+        AC_CHECK_FUNC($cheev, [ax_lapack_ok=yes])
+        LIBS="$save_LIBS"
+fi
+
+# Generic LAPACK library?
+for lapack in lapack lapack_rs6k; do
+        if test $ax_lapack_ok = no; then
+                save_LIBS="$LIBS"; LIBS="$BLAS_LIBS $LIBS"
+                AC_CHECK_LIB($lapack, $cheev,
+                    [ax_lapack_ok=yes; LAPACK_LIBS="-l$lapack"], [], [$FLIBS])
+                LIBS="$save_LIBS"
+        fi
+done
+
+AC_SUBST(LAPACK_LIBS)
+
+# Finally, execute ACTION-IF-FOUND/ACTION-IF-NOT-FOUND:
+if test x"$ax_lapack_ok" = xyes; then
+        ifelse([$1],,AC_DEFINE(HAVE_LAPACK,1,[Define if you have LAPACK library.]),[$1])
+        :
+else
+        ax_lapack_ok=no
+        $2
+fi
+])dnl AX_LAPACK
diff -Nur snfit-2.2.2b.orig/Makefile.am snfit-2.2.2b/Makefile.am
--- snfit-2.2.2b.orig/Makefile.am	2011-02-08 22:01:37.000000000 +0000
+++ snfit-2.2.2b/Makefile.am	2011-02-08 22:01:46.000000000 +0000
@@ -1,4 +1,21 @@
+ACLOCAL_AMFLAGS = -I m4
+CLEANFILES = *~
 
-EXTRA_DIST = mgr/install_cmt mgr/requirements
+MAINTAINERCLEANFILES = \
+	aclocal.m4 \
+	configure \
+	Makefile.in \
+	config.h \
+	config.h.in \
+	autoconf/* \
+	m4/libtool.m4 \
+	m4/lt*.m4
 
+DISTCLEANFILES = snfit.pc
+
+EXTRA_DIST= README NEWS ChangeLog INSTALL COPYING \
+	mgr/install_cmt mgr/requirements
 SUBDIRS = src 
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = snfit.pc
diff -Nur snfit-2.2.2b.orig/snfit.pc.in snfit-2.2.2b/snfit.pc.in
--- snfit-2.2.2b.orig/snfit.pc.in	1970-01-01 01:00:00.000000000 +0100
+++ snfit-2.2.2b/snfit.pc.in	2011-02-08 22:01:46.000000000 +0000
@@ -0,0 +1,13 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: @PACKAGE_NAME@
+Description: Spectral Adaptive Lightcurve Template
+Version: @PACKAGE_VERSION@
+URL: http://supernovae.in2p3.fr/~guy/salt/
+Requires: cfitsio
+Libs: -L${libdir} -lsnfit @LIBS@
+Libs.private: -lm
+Cflags: -I${includedir}/@PACKAGE_NAME@
diff -Nur snfit-2.2.2b.orig/src/Makefile.am snfit-2.2.2b/src/Makefile.am
--- snfit-2.2.2b.orig/src/Makefile.am	2011-02-08 22:01:37.000000000 +0000
+++ snfit-2.2.2b/src/Makefile.am	2011-02-08 22:03:04.000000000 +0000
@@ -1,10 +1,9 @@
 
-AM_CXXFLAGS = -DUSELAPACK -DgFortran @CFITSIOCPPFLAGS@ 
-
-AM_LDFLAGS  = -L$(top_builddir)/src -R @libdir@ -L@libdir@ @CFITSIOLDFLAGS@
-
 lib_LTLIBRARIES = libsnfit.la
 
+libsnfit_la_LIBADD = @CFITSIO_LIBS@ @BLAS_LIBS@ @LAPACK_LIBS@
+libsnfit_la_CPPFLAGS = @CFITSIO_CFLAGS@ @CFORTRAN_CFLAGS@
+
 libsnfit_la_SOURCES = \
 dfact.F         minuit.F  mncrck.F  mneval.F  mnhess.F  mnmatu.F  mnplot.F  mnsave.F  mntiny.F \
 dfeqn.F         mnamin.F  mncros.F  mnexcm.F  mnimpr.F  mnmigr.F  mnpout.F  mnscan.F  mnunpt.F \
@@ -39,7 +38,8 @@
 malmquist_bias.cc \
 wmap5cosmo.cc wmap5cosmoprior.cc
 
-include_HEADERS = \
+pkgincludedir=$(includedir)/@PACKAGE@
+pkginclude_HEADERS = \
  fact.inc  feqn.inc  finv.inc  sfact.inc \
  sfinv.inc d506cm.inc  d506dp.inc \
 brent.h            datafile.h        hubblefitter.h            lctemplate.h      pilot.h             stringlist.h \
@@ -55,18 +55,18 @@
 datacards.h        hostabsorption.h  lcmodels.h                pathutils.h       spectrumtemplate.h \
 bspline.c
 
-
+LDADD = libsnfit.la
 bin_PROGRAMS = snfit snlc snphotoz snmag snstat hubblefit
 
 snfit_SOURCES     = snfit.cc
-snfit_LDFLAGS     = -L$(top_builddir)/src -lsnfit
 snlc_SOURCES      = snlc.cc
-snlc_LDFLAGS      = -L$(top_builddir)/src -lsnfit
 snphotoz_SOURCES  = snphotoz.cc
-snphotoz_LDFLAGS  = -L$(top_builddir)/src -lsnfit
 snmag_SOURCES     = snmag.cc
-snmag_LDFLAGS     = -L$(top_builddir)/src -lsnfit
 snstat_SOURCES    = snstat.cc
-snstat_LDFLAGS    = -L$(top_builddir)/src -lsnfit
 hubblefit_SOURCES = hubblefit.cc
-hubblefit_LDFLAGS = -L$(top_builddir)/src -lsnfit
+
+CLEANFILES = *~
+MAINTAINERCLEANFILES = \
+        Makefile.in \
+        stamp-* \
+        snfit_config.h.in
diff -Nur snfit-2.2.2b.orig/src/matvect.cc snfit-2.2.2b/src/matvect.cc
--- snfit-2.2.2b.orig/src/matvect.cc	2011-02-08 22:01:37.000000000 +0000
+++ snfit-2.2.2b/src/matvect.cc	2011-02-08 22:01:46.000000000 +0000
@@ -5,14 +5,14 @@
 #include <matvect.h>
 #include <fileutils.h>
 
-#ifdef USECFITSIO
+#ifdef HAVE_CFITSIO
 #include <fitsio.h>
 #endif
 
 
 using namespace std; 
 
-#ifdef USECERNLIB
+#ifdef HAVE_CERNLIB
 #define dsinv dsinv_
 #define dfact dfact_
 #define dfinv dfinv_
@@ -31,7 +31,7 @@
 }
 #endif
 
-#ifdef USELAPACK
+#ifdef HAVE_LAPACK
 // using lapack. 
 // Signatures (with a fortran accent) can be obtained on many linux systems from e.g. "man dposv" 
 extern "C" {
@@ -261,6 +261,12 @@
     abort();
   }
 
+  if (A.SizeX() == 0)
+    {
+      cout << " error in matvect.cc, cholesky_solve : Matrix dimension is 0 "<< endl;
+      abort();
+    }
+
   double *a = A.NonConstData();
   double *b = B.NonConstData();
   
@@ -894,6 +900,13 @@
   double beta = 0;
   int ldc = ResultMat.nx;
   
+  if (Right.nx == 0) // we should throw an exception
+    {
+      std::cerr << " in matrix multiplication : Right.nx == 0 cannot work " << std::endl;
+      abort ();
+    }
+	
+
   dgemm_(transa,transb,m,n,k,
 	 alpha, Right.data,lda, 
 	 data, ldb, 
@@ -1039,7 +1052,7 @@
 
 
 
-#ifdef USECFITSIO
+#ifdef HAVE_CFITSIO
 int Mat::readFits(const string &FitsName) {
   
   if( ! FileExists(FitsName) ) {
@@ -1191,7 +1204,7 @@
       }
 } 
 
-#ifdef USECERNLIB
+#ifdef HAVE_CERNLIB
 int Mat::SymMatInvert() {
   if(nx!=ny) {
     cout << "Mat::Symmetrize ERROR nx!=ny nx,ny = " << nx << "," << ny << endl;
@@ -1210,8 +1223,8 @@
 #endif
 
 int Mat::CholeskyInvert(const char *UorLorS) {
-  if(nx!=ny) {
-    cout << "Mat::CholeskyInvert ERROR nx!=ny nx,ny = " << nx << "," << ny << endl;
+  if(nx!=ny || nx == 0) {
+    cout << "Mat::CholeskyInvert ERROR (bad dimensions) nx,ny = " << nx << "," << ny << endl;
     abort();
   }
 
@@ -1238,7 +1251,7 @@
   if (info != 0) 
     {
       cerr << "Mat::CholeskyInvert : could not invert, info = " << info << endl;
-      exit(info);
+      abort();
     }
   
   if(is_symmetric) {
diff -Nur snfit-2.2.2b.orig/src/matvect.h snfit-2.2.2b/src/matvect.h
--- snfit-2.2.2b.orig/src/matvect.h	2011-02-08 22:01:37.000000000 +0000
+++ snfit-2.2.2b/src/matvect.h	2011-02-08 22:01:46.000000000 +0000
@@ -1,6 +1,10 @@
 #ifndef MATVECT__H
 #define MATVECT__H
 
+#ifdef HAVE_CONFIG_H
+#include "snfit_config.h"
+#endif
+
 #include <iostream>
 #include <string>
 
@@ -11,11 +15,11 @@
 class Mat;
 
 // if you link with lapack uncomment this
-//#define USELAPACK
+//#define HAVE_LAPACK
 // if you link with cfitsio uncomment this
-//#define USECFITSIO
+//#define HAVE_CFITSIO
 // if you link with cernlib uncomment this
-#define USECERNLIB
+#define HAVE_CERNLIB
 
 
 
@@ -25,7 +29,7 @@
 // Routines for solving linear systems + inversion 
 //==================================================================
 
-#ifdef USELAPACK
+#ifdef HAVE_LAPACK
 
 // solving linear system A.X = B 
 // Uses lapack dposv_
@@ -160,7 +164,7 @@
   Mat WithoutRows(size_t y_min,size_t y_max) const;
   Mat WithoutColumns(size_t x_min,size_t x_max) const;
 
-#ifdef USECFITSIO
+#ifdef HAVE_CFITSIO
   // i/o in fits for matrices
   int readFits(const std::string &FitsName);
   int writeFits(const std::string &FitsName) const;
@@ -173,7 +177,7 @@
   
   void Symmetrize(const char* UorL); // "L" <=> (*this)(i,j)!=0 for i>=j 
   
-#ifdef USECERNLIB
+#ifdef HAVE_CERNLIB
   // inverts a symetric posdef matrix using DSINV  CERNLIB's routine
   int SymMatInvert();
 #endif
diff -Nur snfit-2.2.2b.orig/src/measureddata.cc snfit-2.2.2b/src/measureddata.cc
--- snfit-2.2.2b.orig/src/measureddata.cc	2011-02-08 22:01:37.000000000 +0000
+++ snfit-2.2.2b/src/measureddata.cc	2011-02-08 22:01:46.000000000 +0000
@@ -1,10 +1,14 @@
+#ifdef HAVE_CONFIG_H
+#include "snfit_config.h"
+#endif
+
 #include "measureddata.h"
 #include "dictfile.h"
 #include "fatalerror.h"
 #include <cmath>
 #include <fstream>
 #include <fstream>
-#ifdef USECFITSIO
+#ifdef HAVE_CFITSIO
 #include <fitsio.h>
 #endif
 
@@ -449,7 +453,7 @@
 }
 #endif
 
-#ifdef USECFITSIO
+#ifdef HAVE_CFITSIO
 static void fitserror(int status) {
   if(status) {
     cerr << " we got these successive errors:" << endl;
@@ -495,7 +499,7 @@
   }
   fits_close_file(fptr, &status); fitserror(status);
 }
-#endif //  USECFITSIO
+#endif //  HAVE_CFITSIO
 
 static void read_an_ascii_array(const string &FileName,Vect &X, Vect &Y,int Ycol) {
   DataFile dfile(FileName);
@@ -1008,7 +1012,7 @@
     
     // check cfitsio
     // ====================================
-#ifndef USECFITSIO
+#ifndef HAVE_CFITSIO
     if(file.HasGlobalKey("Fits_Fluxes") || file.HasGlobalKey("Fits_Errors")) {
       cerr << "FATAL error, current installation of software cannot deal with fits files" << endl;
       exit(-12);
@@ -1022,7 +1026,7 @@
     // ====================================
     if(file.HasGlobalKey("ASCII_Fluxes"))
       read_an_ascii_array(fullpath(FileName,file.GlobalValue("ASCII_Fluxes")),Wavelength,Flux,2);
-#ifdef USECFITSIO
+#ifdef HAVE_CFITSIO
     if(file.HasGlobalKey("Fits_Fluxes"))
       read_a_fits_array(fullpath(FileName,file.GlobalValue("Fits_Fluxes")),Wavelength,Flux);
 #endif
@@ -1037,7 +1041,7 @@
     // ====================================
     if(file.HasGlobalKey("ASCII_Errors"))
       read_an_ascii_array(fullpath(FileName,file.GlobalValue("ASCII_Errors")),Wavelength_for_error,Error,2);
-#ifdef USECFITSIO
+#ifdef HAVE_CFITSIO
     if(file.HasGlobalKey("Fits_Errors"))
       read_a_fits_array(fullpath(FileName,file.GlobalValue("Fits_Errors")),Wavelength_for_error,Error);
 #endif
