--- src/montage2.f.orig	2010-06-23 21:47:29.000000000 +0100
+++ src/montage2.f	2010-06-23 21:48:38.000000000 +0100
@@ -990,7 +990,7 @@
       LIMHI=N
 C
   100 DKEY=DATUM(LIMLO)
-D     TYPE *,LIMLO,LIMHI
+C     TYPE *,LIMLO,LIMHI
 C
 C Compare all elements in the sub-vector between LIMLO and LIMHI with
 C the current key datum.
@@ -1031,11 +1031,11 @@
 C this is where we stick the key back into the vector.
 C
       DATUM(LO)=DKEY
-D     DO 1666 I=LIMLO,LO-1
-D1666 TYPE *,DATUM(I)
-D     TYPE *,DATUM(L0),' KEY'
-D     DO 2666 I=LO+1,LIMHI
-D2666 TYPE *,DATUM(I)
+C     DO 1666 I=LIMLO,LO-1
+C1666 TYPE *,DATUM(I)
+C     TYPE *,DATUM(L0),' KEY'
+C     DO 2666 I=LO+1,LIMHI
+C2666 TYPE *,DATUM(I)
 C
 C At this point in the subroutine, all data between LIMLO and LO-1, 
 C inclusive, are less than DATUM(LO), and all data between LO+1 and 
@@ -1080,8 +1080,8 @@
       STKLO(NSTAK)=LIMLO
       STKHI(NSTAK)=LO-1
       LIMLO=LO+1
-D     DO 3666 I=1,NSTAK
-D3666 TYPE *,' STACK: ',STKLO(I),STKHI(I)
+C     DO 3666 I=1,NSTAK
+C3666 TYPE *,' STACK: ',STKLO(I),STKHI(I)
       GO TO 100
 C
   300 CONTINUE
@@ -1115,8 +1115,8 @@
       STKLO(NSTAK)=LO+1
       STKHI(NSTAK)=LIMHI
       LIMHI=LO-1
-D     DO 4666 I=1,NSTAK
-D4666 TYPE *,' STACK: ',STKLO(I),STKHI(I)
+C     DO 4666 I=1,NSTAK
+C4666 TYPE *,' STACK: ',STKLO(I),STKHI(I)
       GO TO 100
 C
   400 CONTINUE
