--- snfit-2.1.9.orig/src/fileutils.cc	2010-03-16 15:38:24.000000000 +0000
+++ snfit-2.1.9/src/fileutils.cc	2010-03-16 16:59:18.000000000 +0000
@@ -476,7 +476,7 @@
 		  continue;
 		}
 	    }
-	  pos                = strstr (token2,words[count]);
+	  pos                = strstr (const_cast<char*>(token2),words[count]);
 	  if(pos==NULL)      {match=0;break;}
 	  if((count==(wnumber-1)) && (a_pattern[lpattern-1]!='*') ) /*Last word.*/
 	    {
